ros/api/v1/hosts.py

```diff
@@ -429,108 +429,103 @@ def get(self):
         idling_systems = count_per_state(system_queryset, {'state': "Idling"})
 
         non_optimized = system_queryset.filter(
-            or_(
-                System.state == 'Oversized',
-                System.state == 'Undersized',
-                System.state == 'Idling',
-                System.state == 'Under pressure'
-            )
+            System.state.in_(
+                ['Oversized', 'Undersized', 'Idling', 'Under pressure'])
         )
-        non_optimized_count = non_optimized.count()
-
-        # cpu counts
-        total_cpu_issues = sum([
-            len(system.cpu_states) if system.cpu_states else 0
-            for system in non_optimized.filter(System.cpu_states is not None)
-        ])
-        cpu_undersized = non_optimized.filter(System.cpu_states.any(SubStates.CPU_UNDERSIZED.value)).count()
-        cpu_under_pressure = non_optimized.filter(System.cpu_states.any(
-            SubStates.CPU_UNDER_PRESSURE.value)
-        ).count()
-        cpu_oversized = non_optimized.filter(System.cpu_states.any(SubStates.CPU_OVERSIZED.value)).count()
-
-        # memory counts
-        total_memory_issues = sum([
-            len(system.memory_states) if system.memory_states else 0
-            for system in non_optimized.filter(System.memory_states is not None)
-        ])
-        memory_undersized = non_optimized.filter(System.memory_states.any(SubStates.MEMORY_UNDERSIZED.value)).count()
-        memory_pressure = non_optimized.filter(System.memory_states.any(
-            SubStates.MEMORY_UNDER_PRESSURE.value)
-        ).count()
-        memory_oversized = non_optimized.filter(System.memory_states.any(SubStates.MEMORY_OVERSIZED.value)).count()
-
-        # io counts
-        total_io_issues = sum([
-            len(system.io_states) if system.io_states else 0
-            for system in non_optimized.filter(System.io_states is not None)
-        ])
-        io_pressure = non_optimized.filter(System.io_states.any(SubStates.IO_UNDER_PRESSURE.value)).count()
-
-        # FIXME - enable this logic after getting IO states from advisor engine
-        # io_undersized = non_optimized.filter(
-        #    System.io_states.any(SubStates.IO_UNDERSIZED.value)).count()
-        # io_oversized = non_optimized.filter(
-        #    System.io_states.any(SubStates.IO_OVERSIZED.value)).count()
-        io_undersized = -1
-        io_oversized = -1
-
-        all_conditions_count = total_cpu_issues + total_memory_issues + total_io_issues
+        non_optimized_count = 0
+        total_keys = ['cpu', 'memory', 'io']
+        state_keys = ['undersized', 'under_pressure', 'oversized']
+        totals = dict.fromkeys(total_keys, 0)
+        cpu_states_dict = dict.fromkeys(state_keys, 0)
+        memory_states_dict = dict.fromkeys(state_keys, 0)
+        io_states_dict = dict.fromkeys(state_keys, 0)
```

Comment:
```suggestion
        state_keys_init = dict.fromkeys(state_keys, 0)
        cpu_states_dict = memory_states_dict = io_states_dict = state_keys_init
```

Reply:
@saltgen - I intentionally avoided assigning same value to these variables as here I am assigning `dict` objects which come under mutable objects.